src/components/auth/AuthGuard.tsx

```diff
@@ -0,0 +1,24 @@
+"use client";
+
+import { useAuth } from "@/context/auth";
+import { useRouter } from "next/navigation";
+import { useEffect } from "react";
+
+const AuthGuard = ({ children }: { children: any }) => {
+	const { user, loading } = useAuth();
+	const router = useRouter();
+
+	useEffect(() => {
+		if (!loading && !user) {
+			router.push("/login");
+		}
+	}, [user, router, loading]);
+
+	if (loading) {
+		return <div>Loading...</div>;
```

Comment:
A spinner would make a good experience here than just a Loading... text

Reply:
Sounds good. I'll implement that.